tests/components/intellifire/test_config_flow.py

```diff
@@ -1,55 +1,205 @@
+        return_value=["192.168.1.69", "192.168.1.33", "192.168.169"],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+
+    assert result["step_id"] == "pick_device"
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_HOST: "192.168.1.33"}
+    )
+    await hass.async_block_till_done()
+    assert result["step_id"] == "pick_device"
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+
+
+async def test_multi_discovery_cannot_connect(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test for multiple firepalce discovery - involing a pick_device step."""
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
+        return_value=["192.168.1.69", "192.168.1.33", "192.168.169"],
+    ):
+
+        mock_intellifire_config_flow.poll.side_effect = ConnectionError
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "pick_device"
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_HOST: "192.168.1.33"}
+        )
+        await hass.async_block_till_done()
+        assert result2["type"] == RESULT_TYPE_FORM
+        assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_form_cannot_connect_manual_entry(
+    hass: HomeAssistant,
+    mock_intellifire_config_flow: MagicMock,
+    mock_fireplace_finder_single: AsyncMock,
 ) -> None:
     """Test we handle cannot connect error."""
     mock_intellifire_config_flow.poll.side_effect = ConnectionError
 
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": config_entries.SOURCE_USER}
     )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "manual_device_entry"
 
     result2 = await hass.config_entries.flow.async_configure(
         result["flow_id"],
         {
-            "host": "1.1.1.1",
+            CONF_HOST: "1.1.1.1",
         },
     )
 
     assert result2["type"] == RESULT_TYPE_FORM
     assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_picker_already_discovered(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test single firedplace UDP discovery."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "host": "192.168.1.3",
+        },
+        title="Fireplace",
+        unique_id=44444,
+    )
+    entry.add_to_hass(hass)
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
```

Comment:
There are few more patch targets that could be cleaned up to only patch where it is needed.